Automattic/wp-calypso
# Memberships: prevent overwriting product title when default name is used

Related to 248-gh-automattic/gold

## Proposed Changes

* Modify detection of default name for product title so that default names can be used so long as it's not the default for the selected schedule.

## Testing Instructions



* Use the calypso live links and navigate to the monetize features (tools > monetize) for a site with paid memberships plans.
* Open the create plan dialog and change the schedule from monthly to yearly, one-time and then back to monthly. Observe with each change how the title changes.
* Create a monthly plan named "Subscription" and save that plan.
* Edit the plan, and observe that the title is still named "Subscription".
* Change the schedule to "one-time" and then to "monthly" and then to "yearly". Observe how the title does not change.
* Close the edit plan dialog and click create new plan.
* Change the name of the plan. Now try changing the schedule - the title should not change.
* Change the name of the plan to "Subscription", then change the schedule to "one-time" and back to "monthly" or "yearly" - the title should not change.

## Pre-merge Checklist
